interfaces/thirdparty/cp2k.py

```diff
@@ -340,64 +344,87 @@ def parse(key, value, indent=''):
         return inp
 
     def _parsemol(self):
-        #make lines shorter
+        # make lines shorter
         inp = self.settings.input.force_eval.subsys
-        #add cell information
+        # add cell information
         nDim = len(self.molecule.lattice)
         keys = ['A', 'B', 'C']
         periodic = ['X', 'XY', 'XYZ']
-        for iDim in range(0,nDim):
-            inp.cell[keys[iDim]] = "{:} {:} {:}".format(*self.molecule.lattice[iDim])
+        for iDim in range(0, nDim):
+            inp.cell[keys[iDim]] = "{:} {:} {:}".format(
+                *self.molecule.lattice[iDim])
         if nDim > 0:
             inp.cell.periodic = periodic[nDim-1]
 
-        #get block of: symbol coords
+        # get block of: symbol coords
         coord_sec = ""
         for atom in self.molecule:
             coord_sec += "\n"
             coord_sec += (" {:}"*4).format(atom.symbol, *atom.coords)
         inp.coord._h = coord_sec
 
     def get_runscript(self):
-        """
-        Run parallel version of Cp2k using srun.
-        """
+        """Run parallel version of Cp2k using srun."""
         # Try to run cp2k using mpirun and otherwise srun (if available)
-        command_tuple = ('mpirun', 'srun')
-        ret = 'cp2k.popt'
-        for command in command_tuple:
-            try:
-                subprocess.run([command, "--help"], stdout=subprocess.DEVNULL)
-                ret = command + ' cp2k.popt'
-                break
-            except OSError:
-                pass
+        cp2k_command = self.settings.get("executable", "cp2k.popt")
+
+        # Check the executable name
+        available_executables = {
+            # Serial single core testing and debugging
+            "sdbg",
+            # Serial general single core usage
+            "sopt",
+            # Parallel (only OpenMP), single node, multi core
+            "ssmp",
+            # Parallel (only MPI) multi-node testing and debugging
+            "pdbg",
+            # Parallel (only MPI) general usage, no threads
+            "popt",
+            # parallel (MPI + OpenMP) general usage, threading might improve scalability and memory usage
+            "psmp"}
+        suffix = Path(cp2k_command).suffix[1:]
+        if suffix not in available_executables:
+            msg = f"unrecognized cp2k executable: {cp2k_command}"
+            raise RuntimeError(msg)
+
+        # Try to run cp2k MPI binaries using mpirun and otherwise srun (if available)
+        if suffix in set({'sdbg', 'sopt'}):
+            ret = cp2k_command
+        else:
+            for command in ('mpirun', 'srun'):
+                try:
+                    subprocess.run([command, "--help"],
+                                   stdout=subprocess.DEVNULL)
+                    ret = f"{command} {cp2k_command}"
+                    break
+                except OSError:
+                    pass
```

Comment:
It's probably wise to raise a ``RuntimeError`` if the ``break`` statement is never reached:
```
            for command in ('mpirun', 'srun'):
                try:
                    subprocess.run([command, "--help"],
                                   stdout=subprocess.DEVNULL)
                    ret = f"{command} {cp2k_command}"
                    break
                except OSError:
                    pass
            else:
                raise RuntimeError("Neither 'mpirun' nor 'srun' were successfully found")
```